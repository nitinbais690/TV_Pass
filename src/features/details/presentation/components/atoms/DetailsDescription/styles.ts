import { StyleSheet } from 'react-native';
import { appPadding, appFonts, appPaddingValues } from 'core/styles/AppStyles';

export const styles = (appColors: any) => {
    return StyleSheet.create({
        textStyle: {
            color: appColors.secondary,
            fontSize: appFonts.xs,
            fontFamily: appFonts.primary,
            fontWeight: '400',
            marginTop: appPaddingValues.md,
        },
        readMoreStyle: {
            fontSize: appFonts.xs,
            fontFamily: appFonts.semibold,
            color: appColors.caption,
            marginTop: appPadding.xxs(),
        },
        tvReadMoreSize: { width: 100, borderRadius: 5 },
    });
};
