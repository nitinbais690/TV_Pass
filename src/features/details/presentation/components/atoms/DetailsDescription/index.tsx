import React, { useState, useEffect } from 'react';
import { Text, View, Platform, TouchableHighlight } from 'react-native';
import { BorderlessButton } from 'react-native-gesture-handler';
import { useLocalization } from 'contexts/LocalizationContext';
import useAppColors from 'core/presentation/hooks/use-app-colors';
import { appFontStyle } from 'core/styles/AppStyles';
import { styles } from './styles';

export default function DetailsDescription(props: DetailsDescriptionProps) {
    const [briefMode, setBriefMode] = useState(true);
    const { strings } = useLocalization();
    let appColors = useAppColors();
    let descriptionStyles = styles(appColors);
    let fontStyle = appFontStyle;

    useEffect(() => {
        if (props.showLongOnly) {
            setBriefMode(false);
        } else if (props.shortDescription && props.shortDescription.length > 0) {
            setBriefMode(true);
        }
    }, [props.shortDescription, props.showLongOnly]);

    return (
        <View style={props.style}>
            <Text
                numberOfLines={briefMode ? 2 : 99}
                ellipsizeMode={'tail'}
                style={[fontStyle.body3, descriptionStyles.textStyle]}>
                {props.longDescription ? props.longDescription : props.shortDescription}
            </Text>
            {!props.showLongOnly &&
                briefMode &&
                (Platform.isTV ? (
                    <TouchableHighlight
                        style={descriptionStyles.tvReadMoreSize}
                        underlayColor={appColors.brandTint}
                        onPress={() => setBriefMode(false)}>
                        <Text style={descriptionStyles.readMoreStyle}>{strings['content_detail.read_more_lbl']}</Text>
                    </TouchableHighlight>
                ) : (
                    <BorderlessButton onPress={() => setBriefMode(false)}>
                        <Text style={descriptionStyles.readMoreStyle}>{strings['content_detail.read_more_lbl']}</Text>
                    </BorderlessButton>
                ))}
            {!props.showLongOnly &&
                !briefMode &&
                (Platform.isTV ? (
                    <TouchableHighlight
                        style={descriptionStyles.tvReadMoreSize}
                        underlayColor={appColors.brandTint}
                        onPress={() => setBriefMode(true)}>
                        <Text style={descriptionStyles.readMoreStyle}>{strings['content_detail.read_less_lbl']}</Text>
                    </TouchableHighlight>
                ) : (
                    <BorderlessButton onPress={() => setBriefMode(true)}>
                        <Text style={descriptionStyles.readMoreStyle}>{strings['content_detail.read_less_lbl']}</Text>
                    </BorderlessButton>
                ))}
        </View>
    );
}

export interface DetailsDescriptionProps {
    longDescription: string | undefined;
    shortDescription: string | undefined;
    style?: {};
    showLongOnly?: boolean;
}
