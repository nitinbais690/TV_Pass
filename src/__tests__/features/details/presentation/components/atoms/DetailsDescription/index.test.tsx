import React from 'react';
import { render, toJSON } from '@testing-library/react-native';
import { StyleSheet } from 'react-native';
import DetailsDescription from 'features/details/presentation/components/atoms/DetailsDescription';

jest.mock('core/presentation/hooks/use-app-colors');

describe('detailsDescription', () => {
    const shortText: string =
        'The story follows a war of succession among competing claimants for control of the Iron Throne of the Seven Kingdoms.';
    const longText: string =
        "North of the Seven Kingdoms of Westeros, Night's Watch soldiers are attacked by supernatural White Walkers. One soldier escapes but is captured at Castle Winterfell. Eddard \"Ned\" Stark, Warden of the North, executes him for desertion. Later, six orphaned dire wolf pups are found and one given to each Stark sibling, including Ned's bastard son, Jon Snow. In King's Landing, the Seven Kingdoms capital, Jon Arryn, the Hand of the King, dies suddenly.";

    test('render with long description and default style', () => {
        const { container } = render(<DetailsDescription longDescription={longText} shortDescription={''} />);

        expect(container).toBeDefined();
        expect(toJSON(container)).toMatchSnapshot();
    });

    test('render with short description and default style', () => {
        const { container } = render(<DetailsDescription longDescription={''} shortDescription={shortText} />);

        expect(container).toBeDefined();
        expect(toJSON(container)).toMatchSnapshot();
    });

    test('render with both description and default style', () => {
        const { container, queryByText } = render(
            <DetailsDescription longDescription={longText} shortDescription={shortText} />,
        );

        expect(container).toBeDefined();
        expect(queryByText(longText)).toBeDefined();
        expect(queryByText(shortText)).toBeNull();
        expect(toJSON(container)).toMatchSnapshot();
    });

    test('render with custom style', () => {
        const styles = StyleSheet.create({
            container: {
                padding: 16,
            },
        });

        const { container } = render(
            <DetailsDescription longDescription={longText} shortDescription={''} style={styles.container} />,
        );

        expect(container).toBeDefined();
        expect(toJSON(container)).toMatchSnapshot();
    });

    test('render always longOnly', () => {
        const { container } = render(
            <DetailsDescription longDescription={longText} shortDescription={''} showLongOnly={true} />,
        );

        expect(container).toBeDefined();
        expect(toJSON(container)).toMatchSnapshot();
    });
});
