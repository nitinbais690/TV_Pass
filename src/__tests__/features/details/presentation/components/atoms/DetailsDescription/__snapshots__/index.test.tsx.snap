// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`detailsDescription render always longOnly 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View>
    <Text
      ellipsizeMode="tail"
      numberOfLines={99}
      style={
        Array [
          Object {
            "fontFamily": "ProximaNova-Regular",
            "fontSize": 18,
            "fontWeight": "600",
          },
          Object {
            "color": "#FFFFFF",
            "fontFamily": "ProximaNova-Regular",
            "fontSize": 18,
            "fontWeight": "400",
            "marginTop": 22,
          },
        ]
      }
    >
      North of the Seven Kingdoms of Westeros, Night's Watch soldiers are attacked by supernatural White Walkers. One soldier escapes but is captured at Castle Winterfell. Eddard "Ned" Stark, Warden of the North, executes him for desertion. Later, six orphaned dire wolf pups are found and one given to each Stark sibling, including Ned's bastard son, Jon Snow. In King's Landing, the Seven Kingdoms capital, Jon Arryn, the Hand of the King, dies suddenly.
    </Text>
  </View>
</View>
`;

exports[`detailsDescription render with both description and default style 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View>
    <Text
      ellipsizeMode="tail"
      numberOfLines={2}
      style={
        Array [
          Object {
            "fontFamily": "ProximaNova-Regular",
            "fontSize": 18,
            "fontWeight": "600",
          },
          Object {
            "color": "#FFFFFF",
            "fontFamily": "ProximaNova-Regular",
            "fontSize": 18,
            "fontWeight": "400",
            "marginTop": 22,
          },
        ]
      }
    >
      North of the Seven Kingdoms of Westeros, Night's Watch soldiers are attacked by supernatural White Walkers. One soldier escapes but is captured at Castle Winterfell. Eddard "Ned" Stark, Warden of the North, executes him for desertion. Later, six orphaned dire wolf pups are found and one given to each Stark sibling, including Ned's bastard son, Jon Snow. In King's Landing, the Seven Kingdoms capital, Jon Arryn, the Hand of the King, dies suddenly.
    </Text>
    <RNGestureHandlerButton
      activeOpacity={0.3}
      borderless={true}
      collapsable={false}
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Object {
            "opacity": 1,
          },
        ]
      }
    >
      <Text
        style={
          Object {
            "color": "#FFFFFF66",
            "fontFamily": "ProximaNova-Semibold",
            "fontSize": 18,
            "marginTop": "1%",
          }
        }
      >
        Read more
      </Text>
    </RNGestureHandlerButton>
  </View>
</View>
`;

exports[`detailsDescription render with custom style 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "padding": 16,
      }
    }
  >
    <Text
      ellipsizeMode="tail"
      numberOfLines={2}
      style={
        Array [
          Object {
            "fontFamily": "ProximaNova-Regular",
            "fontSize": 18,
            "fontWeight": "600",
          },
          Object {
            "color": "#FFFFFF",
            "fontFamily": "ProximaNova-Regular",
            "fontSize": 18,
            "fontWeight": "400",
            "marginTop": 22,
          },
        ]
      }
    >
      North of the Seven Kingdoms of Westeros, Night's Watch soldiers are attacked by supernatural White Walkers. One soldier escapes but is captured at Castle Winterfell. Eddard "Ned" Stark, Warden of the North, executes him for desertion. Later, six orphaned dire wolf pups are found and one given to each Stark sibling, including Ned's bastard son, Jon Snow. In King's Landing, the Seven Kingdoms capital, Jon Arryn, the Hand of the King, dies suddenly.
    </Text>
    <RNGestureHandlerButton
      activeOpacity={0.3}
      borderless={true}
      collapsable={false}
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Object {
            "opacity": 1,
          },
        ]
      }
    >
      <Text
        style={
          Object {
            "color": "#FFFFFF66",
            "fontFamily": "ProximaNova-Semibold",
            "fontSize": 18,
            "marginTop": "1%",
          }
        }
      >
        Read more
      </Text>
    </RNGestureHandlerButton>
  </View>
</View>
`;

exports[`detailsDescription render with long description and default style 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View>
    <Text
      ellipsizeMode="tail"
      numberOfLines={2}
      style={
        Array [
          Object {
            "fontFamily": "ProximaNova-Regular",
            "fontSize": 18,
            "fontWeight": "600",
          },
          Object {
            "color": "#FFFFFF",
            "fontFamily": "ProximaNova-Regular",
            "fontSize": 18,
            "fontWeight": "400",
            "marginTop": 22,
          },
        ]
      }
    >
      North of the Seven Kingdoms of Westeros, Night's Watch soldiers are attacked by supernatural White Walkers. One soldier escapes but is captured at Castle Winterfell. Eddard "Ned" Stark, Warden of the North, executes him for desertion. Later, six orphaned dire wolf pups are found and one given to each Stark sibling, including Ned's bastard son, Jon Snow. In King's Landing, the Seven Kingdoms capital, Jon Arryn, the Hand of the King, dies suddenly.
    </Text>
    <RNGestureHandlerButton
      activeOpacity={0.3}
      borderless={true}
      collapsable={false}
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Object {
            "opacity": 1,
          },
        ]
      }
    >
      <Text
        style={
          Object {
            "color": "#FFFFFF66",
            "fontFamily": "ProximaNova-Semibold",
            "fontSize": 18,
            "marginTop": "1%",
          }
        }
      >
        Read more
      </Text>
    </RNGestureHandlerButton>
  </View>
</View>
`;

exports[`detailsDescription render with short description and default style 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View>
    <Text
      ellipsizeMode="tail"
      numberOfLines={2}
      style={
        Array [
          Object {
            "fontFamily": "ProximaNova-Regular",
            "fontSize": 18,
            "fontWeight": "600",
          },
          Object {
            "color": "#FFFFFF",
            "fontFamily": "ProximaNova-Regular",
            "fontSize": 18,
            "fontWeight": "400",
            "marginTop": 22,
          },
        ]
      }
    >
      The story follows a war of succession among competing claimants for control of the Iron Throne of the Seven Kingdoms.
    </Text>
    <RNGestureHandlerButton
      activeOpacity={0.3}
      borderless={true}
      collapsable={false}
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Object {
            "opacity": 1,
          },
        ]
      }
    >
      <Text
        style={
          Object {
            "color": "#FFFFFF66",
            "fontFamily": "ProximaNova-Semibold",
            "fontSize": 18,
            "marginTop": "1%",
          }
        }
      >
        Read more
      </Text>
    </RNGestureHandlerButton>
  </View>
</View>
`;
